src/main/java/org/skriptlang/skript/bukkit/loottables/LootTableUtils.java

```diff
@@ -0,0 +1,59 @@
+package org.skriptlang.skript.bukkit.loottables;
+
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockState;
+import org.bukkit.loot.LootTable;
+import org.bukkit.loot.Lootable;
+import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.UnknownNullability;
+
+/**
+ * Utility class for loot tables.
+ */
+public class LootTableUtils {
+
+	/**
+	 * * Checks whether a block or entity is an instance of {@link Lootable}. This is done because a block is not an instance of Lootable, but a block state is.
+	 * @param object the object to check.
+	 * @return whether the object is lootable.
+	 */
+	public static boolean isLootable(Object object) {
+		if (object instanceof Block block)
+			object = block.getState();
+		return object instanceof Lootable;
+	}
+
+	/**
+	 * Gets the Lootable instance of an object.
+	 * @param object the object to get the Lootable instance of.
+	 * @return the Lootable instance of the object.
+	 */
+	public static @UnknownNullability Lootable getAsLootable(Object object) {
```

Comment:
actually, coming back to this, this should be UnknownNullability imo

Reply:
Since you updated the description stating to call isLootable first, you can remove the annotation altogether. Whether you want to return null or throw an exception for invalid inputs is up to you.